packages/venia-concept/src/RootComponents/Product/Product.js

```diff
@@ -51,13 +58,17 @@ class Product extends Component {
                     if (loading) return loadingIndicator;
 
                     const product = data.productDetail.items[0];
-
-                    return (
-                        <ProductFullDetail
-                            product={this.mapProduct(product)}
-                            addToCart={this.props.addItemToCart}
-                        />
-                    );
+                    const productData = this.mapProduct(product);
+                    if (productData) {
+                        return (
+                            <ProductFullDetail
+                                product={this.mapProduct(product)}
+                                addToCart={this.props.addItemToCart}
+                            />
+                        );
+                    } else {
+                        return <div>Product Out Of Stock</div>;
```

Comment:
We have a very basic [ErrorView component](https://github.com/magento-research/pwa-studio/blob/develop/packages/venia-concept/src/components/ErrorView/errorView.js) that would work nicely here. For consistency and ease of update later, let's use that.

Feel free to add an `outOfStock` (boolean) prop to it that shows this message when the prop is `true`.

Then you can just render `<ErrorView outOfStock={true} />` here 👍